munet/base.py

```diff
@@ -949,7 +957,11 @@ def _cmd_status(self, cmds, raises=False, warn=True, stdin=None, **kwargs):
 
         pinput, stdin = Commander._cmd_status_input(stdin)
         p, actual_cmd = self._popen("cmd_status", cmds, stdin=stdin, **kwargs)
-        o, e = p.communicate(pinput, timeout=timeout)
+        if output:
+            o, e = p.communicate(pinput, timeout=timeout)
+        else:
+            o = ''
+            e = ''
         return self._cmd_status_finish(p, cmds, actual_cmd, o, e, raises, warn)
```

Comment:
This call is expecting to run after the process has completed (one side effect of the `p.communicate` call above is that it waits until this happens). If we look at `_cmd_status_finish` you'll see it checks `p.returncode` as a boolean (expecting 0 to mean success); however, in the not-completed-running case `p.returncode` will be `None` so this also looks like success.

Reply:
I figured that in the case that an immediate error occurred within the process, then maybe catching it isn't a bad thing. In retrospect though, leaving that up to chance is probably a bad idea and consistency would be preferred so I will probably modify this to skip the `_cmd_status_finish` entirely.